src/plugins/telemetry/server/plugin.ts

```diff
@@ -110,6 +121,24 @@ export class TelemetryPlugin implements Plugin {
     }
 
     this.fetcherTask.start(core, { telemetryCollectionManager });
+
+    return {
+      getIsOptedIn: async () => {
+        const internalRepository = new SavedObjectsClient(savedObjects.createInternalRepository());
+        const telemetrySavedObject = await getTelemetrySavedObject(internalRepository!);
+        const config = await this.config$.pipe(take(1)).toPromise();
+        const allowChangingOptInStatus = config.allowChangingOptInStatus;
+        const configTelemetryOptIn = typeof config.optIn === 'undefined' ? null : config.optIn;
+        const currentKibanaVersion = this.currentKibanaVersion;
+
+        return getTelemetryOptIn({
```

Comment:
This method has an additional piece of logic that returns `null` when the minor version has changed: https://github.com/Bamieh/kibana/blob/64739fd76f02d9e137584a02866ec689f193b589/src/plugins/telemetry/common/telemetry_config/get_telemetry_opt_in.ts#L55-L57

Is this the behaviour we want in this public API?